desktop/test/run-visdiff.js

```diff
@@ -0,0 +1,185 @@
+          execSync(`cp ${fromFile} ${filenameParts.dir}/${filenameParts.name}-${commit}${filenameParts.ext}`)
+        }
+      }
+
+      if (result === DIFF_CHANGED) {
+        changedResults.push(filenameParts.name)
+      } else if (result === DIFF_NEW) {
+        newResults.push(filenameParts.name)
+      } else if (result === DIFF_REMOVED) {
+        removedResults.push(filenameParts.name)
+      }
+    }
+  })
+
+  const s3Env = {
+    ...process.env,
+    AWS_ACCESS_KEY_ID: process.env['VISDIFF_AWS_ACCESS_KEY_ID'],
+    AWS_SECRET_ACCESS_KEY: process.env['VISDIFF_AWS_SECRET_ACCESS_KEY']
+  }
+  console.log(`Uploading ${diffDir} to ${BUCKET_S3}...`)
+  execSync(`s3cmd put --acl-public -r screenshots/${diffDir} ${BUCKET_S3}`, {env: s3Env})
+  console.log('Screenshots uploaded.')
+
+  var ghClient = github.client(process.env['VISDIFF_GH_TOKEN'])
+  var ghIssue = ghClient.issue('keybase/client', process.env['TRAVIS_PULL_REQUEST'])
+
+  const commentLines = []
+  let imageCount = 0
+
+  newResults.forEach(name => {
+    const afterURL = `${BUCKET_HTTP}/${diffDir}/${name}-${commitRange[1]}.png`
+    const line = ` * Added: **${name}**`
+    if (imageCount > MAX_INLINE_IMAGES) {
+      commentLines.push(line + ` [(view)](${afterURL})`)
+    } else {
+      commentLines.push(line + '  ')
+      commentLines.push(`   ![${name} rendered](${afterURL})`)
+      imageCount++
+    }
+  })
+
+  changedResults.forEach(name => {
+    const diffURL = `${BUCKET_HTTP}/${diffDir}/${name}.png`
+    const beforeURL = `${BUCKET_HTTP}/${diffDir}/${name}-${commitRange[0]}.png`
+    const afterURL = `${BUCKET_HTTP}/${diffDir}/${name}-${commitRange[1]}.png`
+    const line = ` * Changed: **${name}** [(before)](${beforeURL}) [(after)](${afterURL})`
+    if (imageCount > MAX_INLINE_IMAGES) {
+      commentLines.push(line + ` [(diff)](${diffURL})`)
+    } else {
+      commentLines.push(line + '  ')
+      commentLines.push(`   ![${name} comparison](${diffURL})`)
+      imageCount++
+    }
+  })
+
+  removedResults.forEach(name => {
+    const beforeURL = `${BUCKET_HTTP}/${diffDir}/${name}-${commitRange[0]}.png`
+    commentLines.push(` * Removed: **${name}** [(view original)](${beforeURL})`)
+  })
+
+  if (commentLines.length > 0) {
+    commentLines.unshift(':mag_right: :zap: These commits introduced some visual changes:')
+  } else {
+    commentLines.push(':mag_right: No visual changes found as a result of these commits.')
+  }
+
+  const commentBody = commentLines.join('\n')
+  ghIssue.createComment({body: commentBody}, (err, res) => {
+    if (err) {
+      console.log('Failed to post visual diff on GitHub:', err)
+      process.exit(1)
+    }
+    console.log('Posted visual diff on GitHub:', res.html_url)
+  })
+}
+
+if (process.argv.length !== 3) {
+  console.log(`Usage: node ${path.basename(process.argv[1])} COMMIT1..COMMIT2`)
+  process.exit(1)
+}
+
+const commitRange = process.argv[2]
+  .split(/\.{2,3}/)  // TRAVIS gives us ranges like START...END
```

Comment:
I'm 90% sure that for any pushes and pull requests this should be set (re: https://developer.github.com/v3/activity/events/types/#pushevent and https://github.com/travis-ci/travis-core/pull/383). For instance, if it's a single commit, I expect it to be of the form "<master>...commitsha". In fact, I think I can remove the line of code that appends `^` to the first commit, since this range already takes it into account.